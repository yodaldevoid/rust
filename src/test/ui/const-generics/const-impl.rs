// run-pass

#![feature(const_generics)]
//~^ WARN the feature `const_generics` is incomplete and may cause the compiler to crash

#[allow(dead_code)]

trait Trait {
    fn test(&self) -> usize;
}

impl<T, const N: usize> Trait for [T; N] {
    fn test(&self) -> usize {
        self.len()
    }
}

fn main() {}
