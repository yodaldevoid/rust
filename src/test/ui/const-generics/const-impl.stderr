warning: the feature `const_generics` is incomplete and may cause the compiler to crash
  --> $DIR/const-impl.rs:3:12
   |
LL | #![feature(const_generics)]
   |            ^^^^^^^^^^^^^^

